### PR TITLE
fix(boilerplate): upgrade i18n-js

### LINKED ISSUE #2705: Outdated i18n-js package in boilerplate
The package `i18n-js` in the boilerplat code is at version `3.9.2`. The latest version as of writing is `4.4.3` (https://www.npmjs.com/package/i18n-js?activeTab=versions), and the used version is 2 years old. The update requires code changes in the boilerplate. To avoid a migration for each project that starts with Ignite, it would be useful to update this package to the latest one and update the code accordingly.

### PR DESCRIPTION
**Pull request by frankcalise**
## Please verify the following:

- [x] `yarn test` **jest** tests pass with new tests, if relevant
- [x] `yarn lint` **eslint** checks pass with new code, if relevant
- [x] `yarn format:check` **prettier** checks pass with new code, if relevant
- [x] `README.md` (or relevant documentation) has been updated with your changes

## Describe your PR
- Closes #2705
- Follows migration notes from https://github.com/fnando/i18n/discussions/24

### CODE REVIEW

**Review by mazenchami — APPROVED** [review]
been using v4 to prep for my talk. its a much better experience than v3! 🎉

File: boilerplate/app/i18n/i18n.ts [hunk 1]
```diff
@@ -9,14 +9,19 @@ import ko from "./ko"
...
+// https://github.com/fnando/i18n/discussions/24
 
 // to use regional locales use { "en-US": enUS } etc
-i18n.translations = { ar, en, "en-US": en, ko, fr, jp }
-
 const fallbackLocale = "en-US"
+export const i18n = new I18n(
+  { ar, en, "en-US": en, ko, fr, jp },
+  { locale: fallbackLocale, defaultLocale: fallbackLocale, enableFallback: true },
+)
```
Comment (lindboe) [inline boilerplate/app/i18n/i18n.ts:21]: It looks like we also have some docs references to the old format:

Screenshot 2024-08-13 at 9 20 50 AM

Other parts of that doc `Internationalization in Ignite Apps` look outdated too (mention of a JSON file in i18n.translations, line 19-21 isn't what it references anymore)

Comment (frankcalise) [reply boilerplate/app/i18n/i18n.ts:21]: Updated with a link to the docs so it's up-to-date

**Review by lindboe — APPROVED** [review]

**Pull request merged**